### PR TITLE
util: fix trailing spaces for enum and set type

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
According to MySQL document, enum and set type should ignore trailing spaces.
In TiDB:
```
mysql> CREATE TABLE `t1` (
    ->   `a` set(' ','a ','b') NOT NULL
    -> ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
Query OK, 0 rows affected (0.02 sec)

mysql> show create table t1;
+-------+------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                       |
+-------+------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `a` set(' ','a ','b') COLLATE utf8mb4_bin NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
Trailing spaces for ' ' and 'a ' do not be ignore for tidb.
 
### What is changed and how it works?
Add a new format function `OutputFormatForEnum` to deal with this situation.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

 - Possible performance regression

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8718)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -85,3 +86,5 @@ require (
 	sourcegraph.com/sourcegraph/appdash v0.0.0-20180531100431-4c381bd170b4
 	sourcegraph.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
 )
+
+replace github.com/pingcap/parser => github.com/imtbkcat/parser v0.0.0-20181217125542-235b095862ae
```
Comment (eurekaka) [inline go.mod]: `go.mod` and `go.sum` are irrelevant changes?

File: executor/show.go [hunk 2]
```diff
@@ -600,7 +600,7 @@ func (e *ShowExec) fetchShowCreateTable() error {
 						defaultValBinaryLiteral := types.BinaryLiteral(defaultValStr)
 						fmt.Fprintf(&buf, " DEFAULT %s", defaultValBinaryLiteral.ToBitLiteralString(true))
 					} else {
-						fmt.Fprintf(&buf, " DEFAULT '%s'", format.OutputFormat(defaultValStr))
+						fmt.Fprintf(&buf, " DEFAULT '%s'", format.OutputFormatForEnmu(defaultValStr))
```
Comment (eurekaka) [inline executor/show.go]: This only works for `SHOW CREATE TABLE`? Shall we cut off the trailing spaces during creation of the table? i.e, my concern is, when we insert rows into the table, whether `a ` is valid or not?

Comment (imtbkcat) [reply executor/show.go]: ok, i will try to solve it.

**Pull request closed without merging**